src/mme-app/actionHandlers/attachActionHandlers.cpp

```diff
@@ -403,24 +435,26 @@ ActStatus ActionHandlers::auth_response_validate(SM::ControlBlock& cb)
 	log_msg(LOG_DEBUG, "Inside auth_response_validate \n");
 	SM::ControlBlock* controlBlk_p = SubsDataGroupManager::Instance()->findControlBlock(cb.getCBIndex());
 
-	UEContext *ue_ctxt = dynamic_cast<UEContext*>(cb.getPermDataBlock());
+	UEContext *ue_ctxt = static_cast<UEContext*>(cb.getPermDataBlock());
 	if (ue_ctxt == NULL)
 	{
 		log_msg(LOG_DEBUG, "auth_response_validate: ue context or procedure ctxt is NULL \n");
+
+		MmeContextManagerUtils::deleteUEContext(cb.getCBIndex());
```

Comment:
Would it be better to handle it as an halt action handler in the states. We can call one common function and if any change is needed only one place change is needed.

Reply:
Not doing this as an action handler, since the UE context itself is null here, there is nothing much that can be done other than cleaning up the procedure context. 
I have added function macros instead to avoid code duplication. Can you check and let me know if they are fine?